static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -0,0 +1,138 @@
+import {AuthService} from '../providers/auth.service';
+import {ActivatedRoute, Router} from '@angular/router';
+import {User} from 'firebase';
+import {Component, NgZone, OnInit} from '@angular/core';
+import {AngularFireDatabase, AngularFireObject} from 'angularfire2/database';
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+
+
+@Component({
+    selector: 'download-template',
+    templateUrl: './download-template.component.html',
+    styleUrls: ['./download-template.component.css']
+})
+export class DownloadTemplateComponent implements OnInit {
+    user: User = null;
+    valueMap: Object = {};
+    templateDirectoryInfoRef: AngularFireObject<any>;
+    templateRenderInfoRef: AngularFireObject<any>;
+    templateRenderInfo: Observable<any> = null;
+    templateDirectoryInfo: Observable<any> = null;
+    templateVariableNameList: Object = null;
+    targetUrl: String = null;
+
+
+    constructor(
+        private authService: AuthService,
+        private db: AngularFireDatabase,
+        private ngZone: NgZone,
+        private route: ActivatedRoute,
+        private router: Router,
+        private http: HttpClient,
+    ) {
+    }
+
+    ngOnInit() {
+        const component = this;
+        component.authService.onAuthStateChanged(function (auth) {
+            if (auth === null) { // If the user is logged out
+                component.router.navigate(['login']);
+            } else {
+                component.user = component.authService.getAuth().currentUser;
+                component.route.params.subscribe(params => {
+                    component.ngZone.run(() => { // Need to do this using NgZone since we're calling a third party API
+                        // let metadataIsValid = true;
```

Comment:
Commented this and the other metadata code out since it breaks the page. @theresamayo0 can you please reimplement this and test to make sure it behaves as expected (also make sure it does not run into any race conditions)